app/services/submission/trait_score_parser.rb

```diff
@@ -82,4 +149,22 @@ def csv
   def input
     @input ||= File.open(@upload.file.path)
   end
+
+  def detect_replication(header_columns)
+    header_columns.any?{ |column_name| column_name && column_name.index(/rep\d+$/) }
+  end
+
+  def split_to_trait_and_replicate(column_name)
+    replicate_number = 0
+    return ['',0] if column_name.blank?
+    trait_name = column_name.strip.gsub(/rep\d+$/) do |replicate|
```

Comment:
Come on, you needed a comment to explain what it does and my version automagically teaches the reader about named capture in regexp ;) Seriously though, it's a minor thingie, I don't mind leaving it as it is.